README.md

```diff
@@ -20,4 +20,4 @@ Language | Link | Author
 ------------ | ------------ | ---
 Node.js | [namebasehq/exchange-api](https://github.com/namebasehq/exchange-api) | Namebase
 Python | [wy/namebase-exchange-python](https://github.com/wy/namebase-exchange-python) | ~wy
-Go | N/A | You? :o
+Go | [sniperem/namebase](https://github.com/sniperem/namebase) | sniperem
```

Comment:
This is wonderful! Thank you so much for making this. I'll merge in this PR right now, but I have some suggestions for improvements to the linked repo:

1. Correctness: it seems that subscribing to the depth endpoint first gets a snapshot and then starts listening to the feed. This might lead to inconsistent state if the following were to happen:

a) Ask for the depth snapshot and receive it up-to-date with event #100
b) Event #101 occurs the very next millisecond
c) Start listening to websocket events and see event #102
d) Apply the diff from the event starting with event #102
e) Result: #101 never takes effect

Am I correct that the current code would be susceptible to this error? I didn't read too closely but it seemed that this issue existed. The solution would be:

a) Start listening to websocket events.
b) See event #102
c) While continuing to listen to events, ask for a snapshot.
d) Receive websocket events #103, #104 while you're still waiting for the snapshot request. 
Receive one up to date with #102
d) Receive the depth snapshot reply. Look at the buffered websocket events and apply #103, #104.
e) As new events come in, just apply them right away. This buffering technique ensures there is no risk of missing events

2) Small stylistic change: the readme uses "NewFromFloat" but it would be preferable to use "NewFromString" so that there is no risk of any rounding weirdness. Probably okay as-is, but the code is less obviously correct when it uses floats.

Just flagging these two comments for you in case you'd like to improve it, but it's great as is! Thanks!

Reply:
For the first problem, I did subscribe to websocket updates first, then got snapshot:
```
        path := baseWsURL + "/ws/v0/ticker/depth"
	wsConn, _, err := websocket.DefaultDialer.Dial(path, nil)
	if err != nil {
		log.Print("[namebase] failed to establish a websocket connection", err)
		return nil, err
	}

	chDepth := make(chan Depth, 1)

	snapshot, err := nb.GetDepth(pair, 50)
	if err != nil {
		return nil, err
	}
```
while `nb.GetDepth()` is waiting for response, websocket updates is also waiting for reading. Maybe I missed something, I'll think it over.
For the 2nd problem, I'll change it